lib/kafka/protocol/message.rb

```diff
@@ -113,6 +102,37 @@ def self.decode(decoder)
 
       private
 
+      # Offsets may be relative with regards to wrapped message offset, but there are special cases.
+      #
+      # Cases when client will receive corrected offsets:
+      #   - When fetch request is version 0, kafka will correct relative offset on broker side before replying fetch response
+      #   - When messages is stored in 0.9 format on disk (broker configured to do so).
+      #
+      # All other cases, compressed inner messages should have relative offset, with below attributes:
+      #   - The container message should have the 'real' offset
+      #   - The container message's offset should be the 'real' offset of the last message in the compressed batch
+      #   - The first inner message should always have offset = 0
```

Comment:
sorry, I was wrong about this.
It doesn’t always start with 0

Reply:
thanks, amended 👍🏼